layouts/archive/index.js

```diff
@@ -40,17 +41,19 @@ const ArchivePage = ({
   const page = Number(router.query.page) || 1;
   const postsQuantity = totalPosts < 6 ? totalPosts : posts?.length; // 6 per page
   const taxStatementTemplate = `Showing ${postsQuantity} of ${totalPosts} posts`;
-
   const [selectedCategories, setSelectedCategories] = useState([]);
   const [selectedTopics, setSelectedTopics] = useState([]);
 
   useEffect(() => {
-    if (selectedCategories.length <= 0 && selectedTopics.length <= 0) {
-      setPosts(firstPagePosts);
```

Comment:
removing this to force refetching result every time we select / deselect topics and categories, that way we always show the correct result